### PR TITLE
Triton pre mlir enabled sm90

### PR DESCRIPTION
**Pull request by vchiley**
H100 enabled triton_pre_mlir
https://github.com/vchiley/triton/compare/triton_pre_mlir...vchiley:triton:triton_pre_mlir_sm90?expand=1

On a single H100, FLOPS using 1B MPT with torch_crossentropy and microbatchsize=12:

triton: 256_993_451_664_066
triton: 254_201_243_061_146 (with ALiBi enabled)
flash:  227_260_623_999_127 (no ALiBi support)
torch:  117_315_579_905_781 (requires microbatchsize=4)

Note: the resulting MFU numbers are not really good...
Does it run? Yes.
Is it faster than A100s? Yes, about 1.4x faster.
Is it way worse than what we were seeing when we benchmarked it previously? Yes; we [previously](https://www.mosaicml.com/blog/coreweave-nvidia-h100-part-1) saw about 2x faster (with SXM4 H100s).
BUT this is extremely prelim testing with PCIe H100s from Lambda Cloud (SXM4 H100s are faster)

Note: testing on 1x H100 so FSDP is disabled (cpu init)

Issues:
1. with llm-foundry: `loss_fn: fused_crossentropy` does not work; we need to revert back to `loss_fn: torch_crossentropy`
2. the following prints out a bunch of times, but then it runs anyway (note: I think this is a Torch2 + H100 issue):
```
Starting training...
******************************
Config:
enabled_algorithms/GradientClipping: true
node_name: unknown because NODENAME environment variable not set
num_gpus_per_node: 1
num_nodes: 1
rank_zero_seed: 17

******************************
'sm_90' is not a recognized processor for this target (ignoring processor)
'sm_90' is not a recognized processor for this target (ignoring processor)
...
'sm_90' is not a recognized processor for this target (ignoring processor)
'sm_90' is not a recognized processor for this target (ignoring processor)
[batch=1/13400]:
         Train time/epoch: 0
         Train time/batch: 0
         Train time/sample: 0
         Train time/batch_in_epoch: 0
```

tldr: triton is faster than flash, and works with ALiBi

### CODE REVIEW

**Comment by vchiley** [comment]
updted branch so it works with H100s and with A100 (verified on interactive instance and tests passing [here](https://github.com/mosaicml/llm-foundry/pull/260))
